File: utils/parser.py
import os


def lire_fichier(chemin_fichier):
    try:
        with open(chemin_fichier, 'r') as fichier:
            lignes = [ligne.rstrip('\n') for ligne in fichier.readlines()]
        return lignes
    except FileNotFoundError:
        print(f"Le fichier '{chemin_fichier}' n'existe pas.")
        return []


def ecrire_fichier(chemin_fichier, lignes):
    with open(chemin_fichier, 'w') as fichier:
        for ligne in lignes:
            fichier.write(ligne + '\n')


def lister_chemins_absolus_dossier(dossier):
    chemins_absolus = []
    for dossier_racine, _, fichiers in os.walk(dossier):
        for fichier in fichiers:
            chemin_absolu = os.path.abspath(
                os.path.join(dossier_racine, fichier))
            chemins_absolus.append(chemin_absolu)
    return chemins_absolus

File: utils/parser.jl
function lire_fichier(chemin_fichier::AbstractString)
	try
		lignes = String[]
		fichier = open(chemin_fichier, "r")
		for ligne in eachline(fichier)
			push!(lignes, chomp(ligne))
		end
		close(fichier)
		return lignes
	catch e
		if isa(e, IOError) && errno(e) == ENOENT
			println("Le fichier '$chemin_fichier' n'existe pas.")
			return String[]
		else
			rethrow()
		end
	end
end

function ecrire_fichier(chemin_fichier::AbstractString, lignes::Vector{String})
	fichier = open(chemin_fichier, "w")
	for ligne in lignes
		println(fichier, ligne)
	end
	close(fichier)
end

function lister_chemins_absolus_dossier(dossier::AbstractString)
	chemins_absolus = String[]
	for (dossier_racine, sous_dossiers, fichiers) in walkdir(dossier)
		for fichier in fichiers
			chemin_absolu = abspath(joinpath(dossier_racine, fichier))
			push!(chemins_absolus, chemin_absolu)
		end
	end
	return chemins_absolus
end
